vnext/Shared/Threading/BatchingQueueThread.h

```diff
@@ -3,11 +3,41 @@
 
 #pragma once
 
+#include <ReactCommon/CallInvoker.h>
 #include <Shared/BatchingMessageQueueThread.h>
 #include <thread>
 
+namespace facebook::react {
+class Instance;
+}
+
 namespace react::uwp {
 
+struct BatchingQueueThread;
+
+struct BatchingQueueCallInvoker : facebook::react::CallInvoker {
+  BatchingQueueCallInvoker(std::shared_ptr<facebook::react::MessageQueueThread> const &queueThread);
+
+  void ThreadCheck() noexcept;
+  void invokeAsync(std::function<void()> &&func) noexcept override;
+  void EnsureQueue() noexcept;
+  void onBatchComplete() noexcept;
+  void quitSynchronous() noexcept;
+  void PostBatch() noexcept;
+  void invokeSync(std::function<void()> &&func) noexcept override;
+
+ private:
+  std::shared_ptr<facebook::react::MessageQueueThread> m_queueThread;
+
+  using WorkItemQueue = std::vector<std::function<void()>>;
+  std::shared_ptr<WorkItemQueue> m_taskQueue;
+  std::mutex m_mutex;
```

Comment:
I see we already had this but it seems unnecesary to keep a shared_ptr here instead of std::vector directly? it should std::move cleanly anyway